### ISSUE TITLE
Compilation errors in generated embeddable pojos if the embeddable is located in a "synthetic" schema

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#11185]
Given a schema `S1` (but not `S2`), it's possible to declare a "synthetic" schema `S2` as follows:

```xml
<embeddable>
    <schema>S2</schema>
    <name>AB</name>
    <tables>S1\.T1</tables>
    <fields>
        <field><expression>A</expression></field>
        <field><expression>B</expression></field>
    </fields>
</embeddable>
```

The `ABRecord` embeddable record is correctly generated in a `S2` schema, but no POJO is being generated. This is a problem, because the `T1` POJO references the `Ab` POJO with this code, which doesn't compile:

```java
/**
 * Setter for <code>S2.AB</code>.
 */
public void setAb(Ab value) {
    setA(value.getA());
    setB(value.getB());
}

/**
 * Getter for <code>S2.AB</code>.
 */
public Ab getAb() {
    return new Ab(
          getA()
        , getB()
    );
}
```

**Comment by lukaseder** [comment]
The bug is simple. We're checking if the schema has any tables, instead of checking if the schema has any embeddables:

```java
// Correct
if (generateEmbeddables() && database.getEmbeddables(schema).size() > 0)
    generateEmbeddables(schema);

// Wrong
if (generateEmbeddables() && generateInterfaces() && database.getTables(schema).size() > 0)
    generateEmbeddableInterfaces(schema);

// Wrong
if (generateEmbeddables() && generatePojos() && database.getTables(schema).size() > 0)
    generateEmbeddablePojos(schema);
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.5 (#11186)

3.15.0 commit will be included in #11182.
